### PR TITLE
Async API, leverage rq+redis, use docker-compose for service setup

### PR DESCRIPTION
**Pull request by awelzel**

### CODE REVIEW

File: ci-based/config.yml [hunk 1]
```diff
@@ -23,3 +34,30 @@ CPU_SET:
...
+    bench_command: /benchmarker/scripts/tiny-benchmark.sh
+    bench_args: -D -b microbenchmarks/table-ops/simple-key-value.zeek
+
+  - id: micro-table-ops-complex-key-value
+    bench_command: /benchmarker/scripts/tiny-benchmark.sh
+    bench_args: -D -b microbenchmarks/table-ops/complex-key-value.zeek
+
+  - id: micro-table-ops-copy
+    bench_command: /benchmarker/scripts/tiny-benchmark.sh
+    bench_args: -D -b microbenchmarks/table-ops/copy.zeek
```
Comment (awelzel) [inline ci-based/config.yml:73]: Hey @timwoj - I suspect this PR will be somewhat overwhelming, maybe the config.yml is a good place to get a rough idea for the configuration of multiple jobs. The other is storage.py where the database/persistence part part has initially moved.

File: ci-based/settings.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
+# If you want custom worker name
+# NAME = 'worker-1024'
+
+# If you want to use a dictConfig
+# <https://docs.python.org/3/library/logging.config.html#logging-config-dictschema>
+# for more complex/consistent logging requirements.
+DICT_CONFIG = {
```
Comment (timwoj) [inline ci-based/settings.py]: What is this settings file used for? Could it be merged with the previous config file?

File: ci-based/zeek_benchmarker/tasks.py [hunk 3]
```diff
@@ -115,30 +121,29 @@ def unpack_build(
...
+        logger.debug("Unpacking %s in container: %s", build_path, command)
+        self._client.containers.run(
+            image=image,
+            mounts=mounts,
+            auto_remove=True,
+            network_disabled=True,
+            working_dir=source_dir,
+            security_opt=["no-new-privileges"],
+            command=["bash", "-x", "-c", command],
+        )
```
Comment (timwoj) [inline ci-based/zeek_benchmarker/tasks.py:229]: I can see this being better than the docker-compose, plus not having to run `subprocess.run` is good.

**Review by timwoj — APPROVED** [review]
This looks good to me overall. I had one minor question about configuration. If you want to rework that docker-compose file into the code that's fine with me.

File: ci-based/settings.py (hunk 2 above)
Comment (awelzel) [reply ci-based/settings.py]: It's the configuration file of the `rq worker`. I've move this part into config.yml and renamed the file to `rq_worker_settings.py`.

File: ci-based/zeek_benchmarker/tasks.py (hunk 3 above)
Comment (awelzel) [reply ci-based/zeek_benchmarker/tasks.py:229]: I've re-worked running the benchmarker container using the Docker API directly. A bit of a draw-back is that the image which runs the benchmarks now has to be built, but not the end of the world IMO.

**Comment by awelzel** [comment]
> This looks good to me overall. I had one minor question about configuration. If you want to rework that docker-compose file into the code that's fine with me.

@timwoj - feel free to take another looks. I mentioned the locations of the new database in DM if you want to look at some of the results.

**Review by timwoj — APPROVED** [review]

**Pull request merged**